### PR TITLE
Bump logstash version to 7.17.1

### PR DESCRIPTION
**Pull request by naveentatikonda**
Signed-off-by: Naveen Tatikonda <navtat@amazon.com>

### Description
Bumping up logstash version to 7.17.1

### Check List
- [x] All tests pass
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Shivamdhar — COMMENTED** [review]
Just curious, why are we using `7.17.1` v as the latest release is `8.1.0`.

**Comment by naveentatikonda** [comment]
> Just curious, why are we using `7.17.1` v as the latest release is `8.1.0`.

We are having some ecs_compatibility(:v8) issues with logstash version >=8.0.0. So, for now we are incrementing version to 7.17.1

**Review by Shivamdhar — APPROVED** [review]

**Review by vamshin — APPROVED** [review]
LGTM! Thanks

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
   Integration-Test-OpenDistro:
     strategy:
       matrix:
-        logstash: [ "7.15.2", "7.16.3" ]
```
Comment (VijayanB) [inline .github/workflows/CI.yml:61]: IMO, we can remove 7.15.2

**Pull request merged**